tedana workflow fails when using `--t2smap` without `--no-reports`

### Summary

When users provide a T2* map as input to tedana, no S0 map is estimated and the figure-generation step fails.

### Additional Detail

```
Traceback (most recent call last):
  File "/opt/miniconda3/envs/nipreps/bin/tedana", line 8, in <module>
    sys.exit(_main())
  File "/opt/miniconda3/envs/nipreps/lib/python3.10/site-packages/tedana/workflows/tedana.py", line 1077, in _main
    tedana_workflow(**kwargs, tedana_command=tedana_command)
  File "/opt/miniconda3/envs/nipreps/lib/python3.10/site-packages/tedana/workflows/tedana.py", line 1051, in tedana_workflow
    reporting.static_figures.plot_t2star_and_s0(io_generator=io_generator, mask=mask_denoise)
  File "/opt/miniconda3/envs/nipreps/lib/python3.10/site-packages/tedana/reporting/static_figures.py", line 571, in plot_t2star_and_s0
    assert os.path.isfile(s0_img), f"File {s0_img} does not exist"
AssertionError: File /Users/taylor/Documents/datasets/logans-dset/tedana-test2-sub-pilot_task-checkerboard/sub-pilot_task-checkerboard_S0map.nii.gz does not exist
```

### Next Steps

<!--
Thank you for submitting your issue!
If you do not receive a response within a calendar week, please post a comment on this issue to catch our attention.
Some issues may not be resolved right away due to the volunteer nature of the project; thank you for your patience!
-->

--- a/tedana/reporting/data/html/report_body_template.html
+++ b/tedana/reporting/data/html/report_body_template.html
@@ -195,7 +195,6 @@
 </div>
 {% endif %}
 <div class="carpet-plots">
-  <h1>T2* and S0</h1>
   {% if t2starExists %}
   <h2>T2*</h2>
   <div class="carpet-plots-image">

--- a/tedana/reporting/data/html/report_body_template.html
+++ b/tedana/reporting/data/html/report_body_template.html
@@ -180,10 +180,13 @@
     </div>
   </div>
 </div>
+{% if adaptiveMaskExists %}
 <div class="carpet-plots">
   <h1>Adaptive mask</h1>
   <img id="adaptiveMask" src="{{ adaptiveMask }}" style="width: 95%; max-width: 1200px; height: auto;" />
 </div>
+{% endif %}
+{% if t2s0Exists %}
 <div class="carpet-plots">
   <h1>T2* and S0</h1>
   <h2>T2*</h2>
@@ -208,6 +211,7 @@
     <img id="rmseTimeseries" src="{{ rmseTimeseries }}" style="width: 95%; max-width: 1200px; height: auto;" />
   </div>
 </div>
+{% endif %}
 </div>
 <div class="info">
   <h1>Info</h1>

--- a/tedana/reporting/data/html/report_body_template.html
+++ b/tedana/reporting/data/html/report_body_template.html
@@ -176,42 +176,77 @@
   <div class="carpet-wrapper">
     <div class="carpet-buttons">{{ buttons }}</div>
     <div class="carpet-plots-image">
-      <img id="imgCarpetPlot" src="{{ initialCarpet }}" style="width: 95%; max-width: 1200px; height: auto;" />
+      <img
+        id="imgCarpetPlot"
+        src="{{ initialCarpet }}"
+        style="width: 95%; max-width: 1200px; height: auto"
+      />
     </div>
   </div>
 </div>
 {% if adaptiveMaskExists %}
 <div class="carpet-plots">
   <h1>Adaptive mask</h1>
-  <img id="adaptiveMask" src="{{ adaptiveMask }}" style="width: 95%; max-width: 1200px; height: auto;" />
+  <img
+    id="adaptiveMask"
+    src="{{ adaptiveMask }}"
+    style="width: 95%; max-width: 1200px; height: auto"
+  />
 </div>
 {% endif %}
-{% if t2s0Exists %}
 <div class="carpet-plots">
   <h1>T2* and S0</h1>
+  {% if t2starExists %}
   <h2>T2*</h2>
   <div class="carpet-plots-image">
-    <img id="t2starBrainPlot" src="{{ t2starBrainPlot }}" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img
+      id="t2starBrainPlot"
+      src="{{ t2starBrainPlot }}"
+      style="width: 95%; max-width: 1200px; height: auto"
+    />
   </div>
   <div class="carpet-plots-image">
-    <img id="t2starHistogram" src="{{ t2starHistogram }}" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img
+      id="t2starHistogram"
+      src="{{ t2starHistogram }}"
+      style="width: 95%; max-width: 1200px; height: auto"
+    />
   </div>
+  {% endif %} {% if s0Exists %}
   <h2>S0</h2>
   <div class="carpet-plots-image">
-    <img id="s0BrainPlot" src="{{ s0BrainPlot }}" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img
+      id="s0BrainPlot"
+      src="{{ s0BrainPlot }}"
+      style="width: 95%; max-width: 1200px; height: auto"
+    />
   </div>
   <div class="carpet-plots-image">
-    <img id="s0Histogram" src="{{ s0Histogram }}" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img
+      id="s0Histogram"
+      src="{{ s0Histogram }}"
+      style="width: 95%; max-width: 1200px; height: auto"
+    />
   </div>
-  <h2>T2* and S0 model fit (RMSE). (Scaled between 2nd and 98th percentiles)</h2>
+  {% endif %} {% if rmseExists %}
+  <h2>
+    T2* and S0 model fit (RMSE). (Scaled between 2nd and 98th percentiles)
+  </h2>
   <div class="carpet-plots-image">
-    <img id="rmseBrainPlot" src="{{ rmseBrainPlot }}" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img
+      id="rmseBrainPlot"
+      src="{{ rmseBrainPlot }}"
+      style="width: 95%; max-width: 1200px; height: auto"
+    />
   </div>
   <div class="carpet-plots-image">
-    <img id="rmseTimeseries" src="{{ rmseTimeseries }}" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img
+      id="rmseTimeseries"
+      src="{{ rmseTimeseries }}"
+      style="width: 95%; max-width: 1200px; height: auto"
+    />
   </div>
-</div>
-{% endif %}
+  {% endif %}
 </div>
 <div class="info">
   <h1>Info</h1>

--- a/tedana/reporting/data/html/report_body_template.html
+++ b/tedana/reporting/data/html/report_body_template.html
@@ -169,65 +169,65 @@
 </style>
 <div class="content">
   <h1>ICA components</h1>
-  $content
+  {{ content }}
 </div>
 <div class="carpet-plots">
   <h1>Carpet plots</h1>
   <div class="carpet-wrapper">
-    <div class="carpet-buttons">$buttons</div>
+    <div class="carpet-buttons">{{ buttons }}</div>
     <div class="carpet-plots-image">
-      <img id="imgCarpetPlot" src="$initialCarpet" style="width: 95%; max-width: 1200px; height: auto;" />
+      <img id="imgCarpetPlot" src="{{ initialCarpet }}" style="width: 95%; max-width: 1200px; height: auto;" />
     </div>
   </div>
 </div>
 <div class="carpet-plots">
   <h1>Adaptive mask</h1>
-  <img id="adaptiveMask" src="$adaptiveMask" style="width: 95%; max-width: 1200px; height: auto;" />
+  <img id="adaptiveMask" src="{{ adaptiveMask }}" style="width: 95%; max-width: 1200px; height: auto;" />
 </div>
 <div class="carpet-plots">
   <h1>T2* and S0</h1>
   <h2>T2*</h2>
   <div class="carpet-plots-image">
-    <img id="t2starBrainPlot" src="$t2starBrainPlot" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img id="t2starBrainPlot" src="{{ t2starBrainPlot }}" style="width: 95%; max-width: 1200px; height: auto;" />
   </div>
   <div class="carpet-plots-image">
-    <img id="t2starHistogram" src="$t2starHistogram" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img id="t2starHistogram" src="{{ t2starHistogram }}" style="width: 95%; max-width: 1200px; height: auto;" />
   </div>
   <h2>S0</h2>
   <div class="carpet-plots-image">
-    <img id="s0BrainPlot" src="$s0BrainPlot" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img id="s0BrainPlot" src="{{ s0BrainPlot }}" style="width: 95%; max-width: 1200px; height: auto;" />
   </div>
   <div class="carpet-plots-image">
-    <img id="s0Histogram" src="$s0Histogram" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img id="s0Histogram" src="{{ s0Histogram }}" style="width: 95%; max-width: 1200px; height: auto;" />
   </div>
   <h2>T2* and S0 model fit (RMSE). (Scaled between 2nd and 98th percentiles)</h2>
   <div class="carpet-plots-image">
-    <img id="rmseBrainPlot" src="$rmseBrainPlot" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img id="rmseBrainPlot" src="{{ rmseBrainPlot }}" style="width: 95%; max-width: 1200px; height: auto;" />
   </div>
   <div class="carpet-plots-image">
-    <img id="rmseTimeseries" src="$rmseTimeseries" style="width: 95%; max-width: 1200px; height: auto;" />
+    <img id="rmseTimeseries" src="{{ rmseTimeseries }}" style="width: 95%; max-width: 1200px; height: auto;" />
   </div>
 </div>
 </div>
 <div class="info">
   <h1>Info</h1>
-  $info
+  {{ info }}
 </div>
 <div class="about">
   <h1>About tedana</h1>
-  $about
+  {{ about }}
 </div>
 <div class="content references">
   <h1>References</h1>
   <ul>
-    $references
+    {{ references }}
   </ul>
 </div>
 
 <script type="text/javascript">
   function updateCarpetPlot(name) {
     // Create variable to hold the image source
-    let carpetPlot = "./figures/$prefix";
+    let carpetPlot = "./figures/{{ prefix }}";
 
     if (name == "optcom") {
       carpetPlot += "carpet_optcom.svg";
@@ -250,4 +250,4 @@
   }
 </script>
 
-$javascript
+{{ javascript }}

--- a/tedana/reporting/data/html/report_carpet_buttons_template.html
+++ b/tedana/reporting/data/html/report_carpet_buttons_template.html
@@ -1,21 +1,36 @@
-<button class="blue" type="button" onClick="updateCarpetPlot('optcom_nogsr')" style="display:$optcomdisp">
-    Optimally combined with global signal
+<button
+  class="blue"
+  type="button"
+  onClick="updateCarpetPlot('optcom_nogsr')"
+  style="display:{{ optcomdisp }}"
+>
+  Optimally combined with global signal
 </button>
 <button class="blue" type="button" onClick="updateCarpetPlot('optcom')">
-    Optimally combined $optcomname
+  Optimally combined {{ optcomname }}
 </button>
 <button class="yellow" type="button" onClick="updateCarpetPlot('denoised')">
-    Denoised $denoisedname
+  Denoised {{ denoisedname }}
 </button>
-<button class="yellow" type="button" onClick="updateCarpetPlot('denoised_mir')" style="display:$denoiseddisp">
-    Denoised after MIR
+<button
+  class="yellow"
+  type="button"
+  onClick="updateCarpetPlot('denoised_mir')"
+  style="display:{{ denoiseddisp }}"
+>
+  Denoised after MIR
 </button>
 <button class="green" type="button" onClick="updateCarpetPlot('accepted')">
-    Accepted $acceptedname
+  Accepted {{ acceptedname }}
 </button>
-<button class="green" type="button" onClick="updateCarpetPlot('accepted_mir')" style="display:$accepteddisp">
-    Accepted after MIR
+<button
+  class="green"
+  type="button"
+  onClick="updateCarpetPlot('accepted_mir')"
+  style="display:{{ accepteddisp }}"
+>
+  Accepted after MIR
 </button>
 <button class="red" type="button" onClick="updateCarpetPlot('rejected')">
-    Rejected
+  Rejected
 </button>

--- a/tedana/reporting/data/html/report_head_template.html
+++ b/tedana/reporting/data/html/report_head_template.html
@@ -8,15 +8,15 @@
       charset="UTF-8"
     />
     <script
-      src="https://cdn.bokeh.org/bokeh/release/bokeh-$bokehversion.min.js"
+      src="https://cdn.bokeh.org/bokeh/release/bokeh-{{ bokehversion }}.min.js"
       crossorigin="anonymous"
     ></script>
     <script
-      src="https://cdn.bokeh.org/bokeh/release/bokeh-widgets-$bokehversion.min.js"
+      src="https://cdn.bokeh.org/bokeh/release/bokeh-widgets-{{ bokehversion }}.min.js"
       crossorigin="anonymous"
     ></script>
     <script
-      src="https://cdn.bokeh.org/bokeh/release/bokeh-tables-$bokehversion.min.js"
+      src="https://cdn.bokeh.org/bokeh/release/bokeh-tables-{{ bokehversion }}.min.js"
       crossorigin="anonymous"
     ></script>
     <link
@@ -52,7 +52,7 @@
       <a
         href="https://github.com/ME-ICA/tedana"
         class="pure-menu-heading pure-menu-link"
-        >tedana $version</a
+        >tedana {{ version }}</a
       >
       <ul class="pure-menu-list">
         <li class="pure-menu-item">
@@ -65,6 +65,6 @@
   </nav>
 
   <body>
-    $body
+    {{ body }}
   </body>
 </html>

--- a/tedana/reporting/data/html/report_info_table_template.html
+++ b/tedana/reporting/data/html/report_info_table_template.html
@@ -3,46 +3,46 @@
     <span class="title">Tedana command used:</span>
   <pre class="language-python">
       <code class="code_block">
-        $command
+        {{ command }}
       </code>
     </pre>
   </p>
   <table>
     <tr>
       <td class="title">System:</td>
-      <td>$system</td>
+      <td>{{ system }}</td>
     </tr>
     <tr>
       <td class="title">Node:</td>
-      <td>$node</td>
+      <td>{{ node }}</td>
     </tr>
     <tr>
       <td class="title">Release:</td>
-      <td>$release</td>
+      <td>{{ release }}</td>
     </tr>
     <tr>
       <td class="title">System version:</td>
-      <td>$sysversion</td>
+      <td>{{ sysversion }}</td>
     </tr>
     <tr>
       <td class="title">Machine:</td>
-      <td>$machine</td>
+      <td>{{ machine }}</td>
     </tr>
     <tr>
       <td class="title">Processor:</td>
-      <td>$processor</td>
+      <td>{{ processor }}</td>
     </tr>
     <tr>
       <td class="title">Python:</td>
-      <td>$python</td>
+      <td>{{ python }}</td>
     </tr>
     <tr>
       <td class="title">Tedana version:</td>
-      <td>$tedana</td>
+      <td>{{ tedana }}</td>
     </tr>
     <tr>
       <td class="title">Other library versions:</td>
-      <td>$python_libraries</td>
+      <td>{{ python_libraries }}</td>
     </tr>
   </table>
 </div>

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -146,28 +146,63 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     # Initial carpet plot (default one)
     initial_carpet = f"./figures/{prefix}carpet_optcom.svg"
 
-    # Check if images exist in the figures directory
-    figures_dir = os.path.join(os.path.dirname(initial_carpet), "")
+    # Get the figures directory - it's in the same directory as the references file
+    figures_dir = os.path.join(os.path.dirname(references), "figures")
+
+    LGR.info(f"Figures directory: {figures_dir}")
+
+    # List all files in the figures directory
+    if os.path.exists(figures_dir):
+        files_in_figures = os.listdir(figures_dir)
+        LGR.info(f"Files in figures directory: {files_in_figures}")
+    else:
+        files_in_figures = []
+        LGR.warning(f"Figures directory not found at {figures_dir}")
 
     # Adaptive mask image
-    adaptive_mask = f"./figures/{prefix}adaptive_mask.svg"
-    adaptive_mask_exists = os.path.isfile(adaptive_mask.replace("./figures/", figures_dir))
+    adaptive_mask_filename = f"{prefix}adaptive_mask.svg"
+    adaptive_mask = os.path.join(figures_dir, adaptive_mask_filename)
+    adaptive_mask_exists = adaptive_mask_filename in files_in_figures
+    LGR.info(
+        f"Checking for adaptive mask: {adaptive_mask_filename}, exists: {adaptive_mask_exists}"
+    )
+
+    # Check if essential T2* and S0 images exist
+    t2star_brain_filename = f"{prefix}t2star_brain.svg"
+    t2star_histogram_filename = f"{prefix}t2star_histogram.svg"
+    s0_brain_filename = f"{prefix}s0_brain.svg"
+    s0_histogram_filename = f"{prefix}s0_histogram.svg"
+    rmse_brain_filename = f"{prefix}rmse_brain.svg"
+    rmse_timeseries_filename = f"{prefix}rmse_timeseries.svg"
 
     # T2* and S0 images
-    t2star_brain = f"./figures/{prefix}t2star_brain.svg"
-    t2star_histogram = f"./figures/{prefix}t2star_histogram.svg"
-    s0_brain = f"./figures/{prefix}s0_brain.svg"
-    s0_histogram = f"./figures/{prefix}s0_histogram.svg"
-    rmse_brain = f"./figures/{prefix}rmse_brain.svg"
-    rmse_timeseries = f"./figures/{prefix}rmse_timeseries.svg"
+    t2star_brain = os.path.join(figures_dir, t2star_brain_filename)
+    t2star_histogram = os.path.join(figures_dir, t2star_histogram_filename)
+    s0_brain = os.path.join(figures_dir, s0_brain_filename)
+    s0_histogram = os.path.join(figures_dir, s0_histogram_filename)
+    rmse_brain = os.path.join(figures_dir, rmse_brain_filename)
+    rmse_timeseries = os.path.join(figures_dir, rmse_timeseries_filename)
 
-    # Check if T2* and S0 images exist
     t2s0_exists = (
-        os.path.isfile(t2star_brain.replace("./figures/", figures_dir))
-        and os.path.isfile(t2star_histogram.replace("./figures/", figures_dir))
-        and os.path.isfile(s0_brain.replace("./figures/", figures_dir))
-        and os.path.isfile(s0_histogram.replace("./figures/", figures_dir))
+        t2star_brain_filename in files_in_figures
+        and t2star_histogram_filename in files_in_figures
+        and s0_brain_filename in files_in_figures
+        and s0_histogram_filename in files_in_figures
     )
+
+    LGR.info(
+        f"Checking for T2*/S0 files: t2star_brain: {t2star_brain_filename} in files: {t2star_brain_filename in files_in_figures}"
+    )
+    LGR.info(
+        f"Checking for T2*/S0 files: t2star_histogram: {t2star_histogram_filename} in files: {t2star_histogram_filename in files_in_figures}"
+    )
+    LGR.info(
+        f"Checking for T2*/S0 files: s0_brain: {s0_brain_filename} in files: {s0_brain_filename in files_in_figures}"
+    )
+    LGR.info(
+        f"Checking for T2*/S0 files: s0_histogram: {s0_histogram_filename} in files: {s0_histogram_filename in files_in_figures}"
+    )
+    LGR.info(f"T2*/S0 section will be shown: {t2s0_exists}")
 
     # Convert bibtex to html
     references, bibliography = _bib2html(references)

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -161,28 +161,38 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
         f"Checking for adaptive mask: {adaptive_mask_filename}, exists: {adaptive_mask_exists}"
     )
 
-    # Check if essential T2* and S0 images exist
+    # Check for T2* images
     t2star_brain_filename = f"{prefix}t2star_brain.svg"
     t2star_histogram_filename = f"{prefix}t2star_histogram.svg"
-    s0_brain_filename = f"{prefix}s0_brain.svg"
-    s0_histogram_filename = f"{prefix}s0_histogram.svg"
-    rmse_brain_filename = f"{prefix}rmse_brain.svg"
-    rmse_timeseries_filename = f"{prefix}rmse_timeseries.svg"
-
-    # T2* and S0 images for HTML
     t2star_brain = f"./figures/{t2star_brain_filename}"
     t2star_histogram = f"./figures/{t2star_histogram_filename}"
+
+    # Check for S0 images
+    s0_brain_filename = f"{prefix}s0_brain.svg"
+    s0_histogram_filename = f"{prefix}s0_histogram.svg"
     s0_brain = f"./figures/{s0_brain_filename}"
     s0_histogram = f"./figures/{s0_histogram_filename}"
+
+    # Check for RMSE images
+    rmse_brain_filename = f"{prefix}rmse_brain.svg"
+    rmse_timeseries_filename = f"{prefix}rmse_timeseries.svg"
     rmse_brain = f"./figures/{rmse_brain_filename}"
     rmse_timeseries = f"./figures/{rmse_timeseries_filename}"
 
-    t2s0_exists = (
-        t2star_brain_filename in files_in_figures
-        and t2star_histogram_filename in files_in_figures
-        and s0_brain_filename in files_in_figures
-        and s0_histogram_filename in files_in_figures
+    # Check if each set of images exists
+    t2star_exists = (
+        t2star_brain_filename in files_in_figures and t2star_histogram_filename in files_in_figures
     )
+
+    s0_exists = s0_brain_filename in files_in_figures and s0_histogram_filename in files_in_figures
+
+    rmse_exists = (
+        rmse_brain_filename in files_in_figures and rmse_timeseries_filename in files_in_figures
+    )
+
+    LGR.info(f"T2* files exist: {t2star_exists}")
+    LGR.info(f"S0 files exist: {s0_exists}")
+    LGR.info(f"RMSE files exist: {rmse_exists}")
 
     # Convert bibtex to html
     references, bibliography = _bib2html(references)
@@ -203,11 +213,13 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
         adaptiveMaskExists=adaptive_mask_exists,
         t2starBrainPlot=t2star_brain,
         t2starHistogram=t2star_histogram,
+        t2starExists=t2star_exists,
         s0BrainPlot=s0_brain,
         s0Histogram=s0_histogram,
-        t2s0Exists=t2s0_exists,
+        s0Exists=s0_exists,
         rmseBrainPlot=rmse_brain,
         rmseTimeseries=rmse_timeseries,
+        rmseExists=rmse_exists,
         references=references,
         javascript=bokeh_js,
         buttons=buttons,

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -191,16 +191,20 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     )
 
     LGR.info(
-        f"Checking for T2*/S0 files: t2star_brain: {t2star_brain_filename} in files: {t2star_brain_filename in files_in_figures}"
+        f"Checking for T2*/S0 files: t2star_brain: {t2star_brain_filename} in files: "
+        f"{t2star_brain_filename in files_in_figures}"
     )
     LGR.info(
-        f"Checking for T2*/S0 files: t2star_histogram: {t2star_histogram_filename} in files: {t2star_histogram_filename in files_in_figures}"
+        f"Checking for T2*/S0 files: t2star_histogram: {t2star_histogram_filename} in files: "
+        f"{t2star_histogram_filename in files_in_figures}"
     )
     LGR.info(
-        f"Checking for T2*/S0 files: s0_brain: {s0_brain_filename} in files: {s0_brain_filename in files_in_figures}"
+        f"Checking for T2*/S0 files: s0_brain: {s0_brain_filename} in files: "
+        f"{s0_brain_filename in files_in_figures}"
     )
     LGR.info(
-        f"Checking for T2*/S0 files: s0_histogram: {s0_histogram_filename} in files: {s0_histogram_filename in files_in_figures}"
+        f"Checking for T2*/S0 files: s0_histogram: {s0_histogram_filename} in files: "
+        f"{s0_histogram_filename in files_in_figures}"
     )
     LGR.info(f"T2*/S0 section will be shown: {t2s0_exists}")
 

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -77,7 +77,14 @@ def _inline_citations(text, bibliography):
 def _get_template_env():
     """Create and return Jinja2 environment with template directory."""
     resource_path = Path(__file__).resolve().parent.joinpath("data", "html")
-    template_env = Environment(loader=FileSystemLoader(str(resource_path)))
+
+    # Use variable_start_string and variable_end_string to match the syntax used in the templates
+    # This allows us to keep the existing templates without modifying them
+    template_env = Environment(
+        loader=FileSystemLoader(str(resource_path)),
+        variable_start_string="${",  # Match the string.Template syntax
+        variable_end_string="}",
+    )
     return template_env
 
 
@@ -197,6 +204,7 @@ def _save_as_html(body):
     head_template = template_env.get_template("report_head_template.html")
 
     html = head_template.render(version=__version__, bokehversion=bokehversion, body=body)
+
     return html
 
 

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -147,17 +147,11 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     initial_carpet = f"./figures/{prefix}carpet_optcom.svg"
 
     # Get the figures directory - it's in the same directory as the references file
-    figures_dir = os.path.join(os.path.dirname(references), "figures")
-
-    LGR.info(f"Figures directory: {figures_dir}")
+    figures_dir = "./figures"
 
     # List all files in the figures directory
-    if os.path.exists(figures_dir):
-        files_in_figures = os.listdir(figures_dir)
-        LGR.info(f"Files in figures directory: {files_in_figures}")
-    else:
-        files_in_figures = []
-        LGR.warning(f"Figures directory not found at {figures_dir}")
+    files_in_figures = os.listdir(figures_dir)
+    LGR.info(f"Files in figures directory: {files_in_figures}")
 
     # Adaptive mask image
     adaptive_mask_filename = f"{prefix}adaptive_mask.svg"

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -146,16 +146,16 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     # Initial carpet plot (default one)
     initial_carpet = f"./figures/{prefix}carpet_optcom.svg"
 
-    # Get the figures directory
-    figures_dir = "figures"
+    # Get the figures directory - relative to the directory containing the references file
+    base_dir = os.path.dirname(references)
+    figures_dir = os.path.join(base_dir, "figures")
 
     # List all files in the figures directory
-    files_in_figures = os.listdir(os.path.abspath(figures_dir))
-    LGR.info(f"Files in figures directory: {files_in_figures}")
+    files_in_figures = os.listdir(figures_dir)
 
     # Adaptive mask image
     adaptive_mask_filename = f"{prefix}adaptive_mask.svg"
-    adaptive_mask = os.path.join(figures_dir, adaptive_mask_filename)
+    adaptive_mask = f"./figures/{adaptive_mask_filename}"
     adaptive_mask_exists = adaptive_mask_filename in files_in_figures
     LGR.info(
         f"Checking for adaptive mask: {adaptive_mask_filename}, exists: {adaptive_mask_exists}"
@@ -169,13 +169,13 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     rmse_brain_filename = f"{prefix}rmse_brain.svg"
     rmse_timeseries_filename = f"{prefix}rmse_timeseries.svg"
 
-    # T2* and S0 images
-    t2star_brain = os.path.join(figures_dir, t2star_brain_filename)
-    t2star_histogram = os.path.join(figures_dir, t2star_histogram_filename)
-    s0_brain = os.path.join(figures_dir, s0_brain_filename)
-    s0_histogram = os.path.join(figures_dir, s0_histogram_filename)
-    rmse_brain = os.path.join(figures_dir, rmse_brain_filename)
-    rmse_timeseries = os.path.join(figures_dir, rmse_timeseries_filename)
+    # T2* and S0 images for HTML
+    t2star_brain = f"./figures/{t2star_brain_filename}"
+    t2star_histogram = f"./figures/{t2star_histogram_filename}"
+    s0_brain = f"./figures/{s0_brain_filename}"
+    s0_histogram = f"./figures/{s0_histogram_filename}"
+    rmse_brain = f"./figures/{rmse_brain_filename}"
+    rmse_timeseries = f"./figures/{rmse_timeseries_filename}"
 
     t2s0_exists = (
         t2star_brain_filename in files_in_figures
@@ -183,24 +183,6 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
         and s0_brain_filename in files_in_figures
         and s0_histogram_filename in files_in_figures
     )
-
-    LGR.info(
-        f"Checking for T2*/S0 files: t2star_brain: {t2star_brain_filename} in files: "
-        f"{t2star_brain_filename in files_in_figures}"
-    )
-    LGR.info(
-        f"Checking for T2*/S0 files: t2star_histogram: {t2star_histogram_filename} in files: "
-        f"{t2star_histogram_filename in files_in_figures}"
-    )
-    LGR.info(
-        f"Checking for T2*/S0 files: s0_brain: {s0_brain_filename} in files: "
-        f"{s0_brain_filename in files_in_figures}"
-    )
-    LGR.info(
-        f"Checking for T2*/S0 files: s0_histogram: {s0_histogram_filename} in files: "
-        f"{s0_histogram_filename in files_in_figures}"
-    )
-    LGR.info(f"T2*/S0 section will be shown: {t2s0_exists}")
 
     # Convert bibtex to html
     references, bibliography = _bib2html(references)

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -9,7 +9,7 @@ from pathlib import Path
 import pandas as pd
 from bokeh import __version__ as bokehversion
 from bokeh import embed, layouts, models
-from jinja2 import Environment, FileSystemLoader
+from jinja2 import Environment, FileSystemLoader, Template
 from pybtex.database.input import bibtex
 from pybtex.plugin import find_plugin
 
@@ -78,13 +78,8 @@ def _get_template_env():
     """Create and return Jinja2 environment with template directory."""
     resource_path = Path(__file__).resolve().parent.joinpath("data", "html")
 
-    # Use variable_start_string and variable_end_string to match the syntax used in the templates
-    # This allows us to keep the existing templates without modifying them
-    template_env = Environment(
-        loader=FileSystemLoader(str(resource_path)),
-        variable_start_string="${",  # Match the string.Template syntax
-        variable_end_string="}",
-    )
+    # Standard Jinja2 environment - no need for special configuration now
+    template_env = Environment(loader=FileSystemLoader(str(resource_path)))
     return template_env
 
 

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -150,7 +150,7 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     figures_dir = "figures"
 
     # List all files in the figures directory
-    files_in_figures = os.listdir(figures_dir)
+    files_in_figures = os.listdir(os.path.abspath(figures_dir))
     LGR.info(f"Files in figures directory: {files_in_figures}")
 
     # Adaptive mask image

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -9,7 +9,7 @@ from pathlib import Path
 import pandas as pd
 from bokeh import __version__ as bokehversion
 from bokeh import embed, layouts, models
-from jinja2 import Environment, FileSystemLoader, Template
+from jinja2 import Environment, FileSystemLoader
 from pybtex.database.input import bibtex
 from pybtex.plugin import find_plugin
 

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -146,8 +146,12 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     # Initial carpet plot (default one)
     initial_carpet = f"./figures/{prefix}carpet_optcom.svg"
 
+    # Check if images exist in the figures directory
+    figures_dir = os.path.join(os.path.dirname(initial_carpet), "")
+
     # Adaptive mask image
     adaptive_mask = f"./figures/{prefix}adaptive_mask.svg"
+    adaptive_mask_exists = os.path.isfile(adaptive_mask.replace("./figures/", figures_dir))
 
     # T2* and S0 images
     t2star_brain = f"./figures/{prefix}t2star_brain.svg"
@@ -156,6 +160,14 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     s0_histogram = f"./figures/{prefix}s0_histogram.svg"
     rmse_brain = f"./figures/{prefix}rmse_brain.svg"
     rmse_timeseries = f"./figures/{prefix}rmse_timeseries.svg"
+
+    # Check if T2* and S0 images exist
+    t2s0_exists = (
+        os.path.isfile(t2star_brain.replace("./figures/", figures_dir))
+        and os.path.isfile(t2star_histogram.replace("./figures/", figures_dir))
+        and os.path.isfile(s0_brain.replace("./figures/", figures_dir))
+        and os.path.isfile(s0_histogram.replace("./figures/", figures_dir))
+    )
 
     # Convert bibtex to html
     references, bibliography = _bib2html(references)
@@ -173,10 +185,12 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
         prefix=prefix,
         initialCarpet=initial_carpet,
         adaptiveMask=adaptive_mask,
+        adaptiveMaskExists=adaptive_mask_exists,
         t2starBrainPlot=t2star_brain,
         t2starHistogram=t2star_histogram,
         s0BrainPlot=s0_brain,
         s0Histogram=s0_histogram,
+        t2s0Exists=t2s0_exists,
         rmseBrainPlot=rmse_brain,
         rmseTimeseries=rmse_timeseries,
         references=references,

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -146,8 +146,8 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     # Initial carpet plot (default one)
     initial_carpet = f"./figures/{prefix}carpet_optcom.svg"
 
-    # Get the figures directory - it's in the same directory as the references file
-    figures_dir = "./figures"
+    # Get the figures directory
+    figures_dir = "figures"
 
     # List all files in the figures directory
     files_in_figures = os.listdir(figures_dir)

--- a/tedana/reporting/html_report.py
+++ b/tedana/reporting/html_report.py
@@ -5,11 +5,11 @@ import os
 import re
 from os.path import join as opj
 from pathlib import Path
-from string import Template
 
 import pandas as pd
 from bokeh import __version__ as bokehversion
 from bokeh import embed, layouts, models
+from jinja2 import Environment, FileSystemLoader
 from pybtex.database.input import bibtex
 from pybtex.plugin import find_plugin
 
@@ -74,9 +74,14 @@ def _inline_citations(text, bibliography):
     return updated_text
 
 
-def _generate_buttons(out_dir, io_generator):
+def _get_template_env():
+    """Create and return Jinja2 environment with template directory."""
     resource_path = Path(__file__).resolve().parent.joinpath("data", "html")
+    template_env = Environment(loader=FileSystemLoader(str(resource_path)))
+    return template_env
 
+
+def _generate_buttons(out_dir, io_generator):
     images_list = [img for img in os.listdir(out_dir) if ".svg" in img]
     optcom_nogsr_disp = "none"
     optcom_name = ""
@@ -96,12 +101,10 @@ def _generate_buttons(out_dir, io_generator):
         accepted_mir_disp = "block"
         accepted_name = "before MIR"
 
-    buttons_template_name = "report_carpet_buttons_template.html"
-    buttons_template_path = resource_path.joinpath(buttons_template_name)
-    with open(str(buttons_template_path)) as buttons_file:
-        buttons_tpl = Template(buttons_file.read())
+    template_env = _get_template_env()
+    template = template_env.get_template("report_carpet_buttons_template.html")
 
-    buttons_html = buttons_tpl.substitute(
+    buttons_html = template.render(
         optcomdisp=optcom_nogsr_disp,
         denoiseddisp=denoised_mir_disp,
         accepteddisp=accepted_mir_disp,
@@ -138,8 +141,6 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     -------
     HTMLReport : an instance of a populated HTML report
     """
-    resource_path = Path(__file__).resolve().parent.joinpath("data", "html")
-
     # Initial carpet plot (default one)
     initial_carpet = f"./figures/{prefix}carpet_optcom.svg"
 
@@ -160,12 +161,10 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     # Update inline citations
     about = _inline_citations(about, bibliography)
 
-    body_template_name = "report_body_template.html"
-    body_template_path = resource_path.joinpath(body_template_name)
-    with open(str(body_template_path)) as body_file:
-        body_tpl = Template(body_file.read())
+    template_env = _get_template_env()
+    body_template = template_env.get_template("report_body_template.html")
 
-    body = body_tpl.substitute(
+    body = body_template.render(
         content=bokeh_id,
         info=info_table,
         about=about,
@@ -194,29 +193,22 @@ def _save_as_html(body):
     body : str
         Body for HTML report with embedded figures
     """
-    resource_path = Path(__file__).resolve().parent.joinpath("data", "html")
-    head_template_name = "report_head_template.html"
-    head_template_path = resource_path.joinpath(head_template_name)
-    with open(str(head_template_path)) as head_file:
-        head_tpl = Template(head_file.read())
+    template_env = _get_template_env()
+    head_template = template_env.get_template("report_head_template.html")
 
-    html = head_tpl.substitute(version=__version__, bokehversion=bokehversion, body=body)
+    html = head_template.render(version=__version__, bokehversion=bokehversion, body=body)
     return html
 
 
 def _generate_info_table(info_dict):
     """Generate a table with relevant information about the system and tedana."""
-    resource_path = Path(__file__).resolve().parent.joinpath("data", "html")
-
-    info_template_name = "report_info_table_template.html"
-    info_template_path = resource_path.joinpath(info_template_name)
-    with open(str(info_template_path)) as info_file:
-        info_tpl = Template(info_file.read())
-
     info_dict = info_dict["GeneratedBy"][0]
     node_dict = info_dict["Node"]
 
-    info_html = info_tpl.substitute(
+    template_env = _get_template_env()
+    info_template = template_env.get_template("report_info_table_template.html")
+
+    info_html = info_template.render(
         command=info_dict["Command"],
         system=node_dict["System"],
         node=node_dict["Name"],

--- a/tedana/reporting/static_figures.py
+++ b/tedana/reporting/static_figures.py
@@ -578,7 +578,7 @@ def plot_t2star_and_s0(
     # Check if S0 image exists, issue warning if not
     s0_exists = os.path.isfile(s0_img)
     if not s0_exists:
-        warnings.warn(f"File {s0_img} does not exist. S0 plots will not be generated.")
+        LGR.info(f"File {s0_img} does not exist. S0 plots will not be generated.")
 
     # Plot histograms
     t2star_data = masking.apply_mask(t2star_img, mask_img)

--- a/tedana/reporting/static_figures.py
+++ b/tedana/reporting/static_figures.py
@@ -574,7 +574,11 @@ def plot_t2star_and_s0(
     s0_img = io_generator.get_name("s0 img")
     mask_img = io.new_nii_like(io_generator.reference_img, mask.astype(int))
     assert os.path.isfile(t2star_img), f"File {t2star_img} does not exist"
-    assert os.path.isfile(s0_img), f"File {s0_img} does not exist"
+
+    # Check if S0 image exists, issue warning if not
+    s0_exists = os.path.isfile(s0_img)
+    if not s0_exists:
+        warnings.warn(f"File {s0_img} does not exist. S0 plots will not be generated.")
 
     # Plot histograms
     t2star_data = masking.apply_mask(t2star_img, mask_img)
@@ -590,18 +594,20 @@ def plot_t2star_and_s0(
     fig.tight_layout()
     fig.savefig(os.path.join(io_generator.out_dir, "figures", t2star_histogram))
 
-    s0_data = masking.apply_mask(s0_img, mask_img)
-    s0_p02, s0_p98 = np.percentile(s0_data, [2, 98])
-    s0_histogram = f"{io_generator.prefix}s0_histogram.svg"
+    # Only plot S0 data if the file exists
+    if s0_exists:
+        s0_data = masking.apply_mask(s0_img, mask_img)
+        s0_p02, s0_p98 = np.percentile(s0_data, [2, 98])
+        s0_histogram = f"{io_generator.prefix}s0_histogram.svg"
 
-    fig, ax = plt.subplots(figsize=(10, 6))
-    ax.hist(s0_data[s0_data <= s0_p98], bins=100)
-    ax.set_xlim(0, s0_p98)
-    ax.set_title("S0", fontsize=20)
-    ax.set_ylabel("Count", fontsize=16)
-    ax.set_xlabel("Arbitrary Units\n(limited to 98th percentile)", fontsize=16)
-    fig.tight_layout()
-    fig.savefig(os.path.join(io_generator.out_dir, "figures", s0_histogram))
+        fig, ax = plt.subplots(figsize=(10, 6))
+        ax.hist(s0_data[s0_data <= s0_p98], bins=100)
+        ax.set_xlim(0, s0_p98)
+        ax.set_title("S0", fontsize=20)
+        ax.set_ylabel("Count", fontsize=16)
+        ax.set_xlabel("Arbitrary Units\n(limited to 98th percentile)", fontsize=16)
+        fig.tight_layout()
+        fig.savefig(os.path.join(io_generator.out_dir, "figures", s0_histogram))
 
     # Plot T2* and S0 maps
     t2star_plot = f"{io_generator.prefix}t2star_brain.svg"
@@ -620,21 +626,25 @@ def plot_t2star_and_s0(
             output_file=os.path.join(io_generator.out_dir, "figures", t2star_plot),
         )
 
-    s0_plot = f"{io_generator.prefix}s0_brain.svg"
-    with warnings.catch_warnings():
-        warnings.filterwarnings("ignore", message="A non-diagonal affine.*", category=UserWarning)
-        plotting.plot_stat_map(
-            s0_img,
-            bg_img=None,
-            display_mode="mosaic",
-            symmetric_cbar=False,
-            black_bg=True,
-            cmap="gray",
-            vmin=s0_p02,
-            vmax=s0_p98,
-            annotate=False,
-            output_file=os.path.join(io_generator.out_dir, "figures", s0_plot),
-        )
+    # Only plot S0 map if the file exists
+    if s0_exists:
+        s0_plot = f"{io_generator.prefix}s0_brain.svg"
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore", message="A non-diagonal affine.*", category=UserWarning
+            )
+            plotting.plot_stat_map(
+                s0_img,
+                bg_img=None,
+                display_mode="mosaic",
+                symmetric_cbar=False,
+                black_bg=True,
+                cmap="gray",
+                vmin=s0_p02,
+                vmax=s0_p98,
+                annotate=False,
+                output_file=os.path.join(io_generator.out_dir, "figures", s0_plot),
+            )
 
 
 def plot_rmse(

--- a/tedana/reporting/static_figures.py
+++ b/tedana/reporting/static_figures.py
@@ -578,7 +578,9 @@ def plot_t2star_and_s0(
     # Check if S0 image exists, issue warning if not
     s0_exists = os.path.isfile(s0_img)
     if not s0_exists:
-        LGR.info(f"File {s0_img} does not exist. S0 plots will not be generated.")
+        LGR.info(
+            "S0 maps and T2* fit metrics are not in report since a pre-existing T2* map was provided"
+        )
 
     # Plot histograms
     t2star_data = masking.apply_mask(t2star_img, mask_img)

--- a/tedana/reporting/static_figures.py
+++ b/tedana/reporting/static_figures.py
@@ -579,8 +579,8 @@ def plot_t2star_and_s0(
     s0_exists = os.path.isfile(s0_img)
     if not s0_exists:
         LGR.info(
-            "S0 maps and T2* fit metrics are not in report since a pre-existing T2* map ",
-            "was provided",
+            "S0 maps and T2* fit metrics are not in report since a pre-existing "
+            "T2* map was provided"
         )
 
     # Plot histograms

--- a/tedana/reporting/static_figures.py
+++ b/tedana/reporting/static_figures.py
@@ -579,7 +579,8 @@ def plot_t2star_and_s0(
     s0_exists = os.path.isfile(s0_img)
     if not s0_exists:
         LGR.info(
-            "S0 maps and T2* fit metrics are not in report since a pre-existing T2* map was provided"
+            "S0 maps and T2* fit metrics are not in report since a pre-existing T2* map ",
+            "was provided",
         )
 
     # Plot histograms

--- a/tedana/reporting/static_figures.py
+++ b/tedana/reporting/static_figures.py
@@ -575,7 +575,7 @@ def plot_t2star_and_s0(
     mask_img = io.new_nii_like(io_generator.reference_img, mask.astype(int))
     assert os.path.isfile(t2star_img), f"File {t2star_img} does not exist"
 
-    # Check if S0 image exists, issue warning if not
+    # Check if S0 image exists, add message to log if not
     s0_exists = os.path.isfile(s0_img)
     if not s0_exists:
         LGR.info(

--- a/tedana/tests/test_integration.py
+++ b/tedana/tests/test_integration.py
@@ -254,6 +254,8 @@ def test_integration_three_echo(skip_integration):
         "-f",
         "--mix",
         os.path.join(out_dir, "desc-ICA_mixing.tsv"),
+        "--t2smap",
+        os.path.join(out_dir, "T2starmap.nii.gz"),
     ]
     tedana_cli._main(args)
 

